app/models/topic.rb

```diff
@@ -126,6 +126,11 @@ def assign(user_id=2, assigned_to)
     self.save
   end
 
+  def self.bulk_assign(post_attributes, assigned_to)
+    Post.create(post_attributes)
+    self.update_all(assigned_user_id: assigned_to, current_status: 'pending')
```

Comment:
I don't know what happens if any one of the posts fails to create (for any reason) - it would be good to think about what happens / what should happen in this case, perhaps?